lib/fastlane/actions/badge.rb

```diff
@@ -0,0 +1,54 @@
+module Fastlane
+  module Actions
+    class BadgeAction < Action
+      def self.run(params)
+        require 'badge'
+        Badge::Runner.new.run('.', params[:dark], params[:custom])
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.description
+        "Automatically add a badge to your iOS app icon"
+      end
+
+      def self.details
+        [
+          "This action will add a light/dark beta badge onto your app icon.",
+          "You can also provide your custom badge/overlay more info:",
+          "https://github.com/HazAT/badge"
+        ].join("\n")
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :dark,
+                                       env_name: "FL_BADGE_DARK",
+                                       description: "adds a dark flavored badge ontop of your icon",
+                                       optional: true,
+                                       is_string: false,
+                                       verify_block: proc do |value|
+                                          raise "dark is only a flag and should always be true".red unless (value == true)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :custom,
+                                       env_name: "FL_BADGE_CUSTOM",
+                                       description: "add your custom overlay/badge image",
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                          raise "custom should be a valid file path".red unless (value and File.exist?(value))
```

Comment:
Use 2 (not 3) spaces for indentation.<br>Don't use parentheses around the condition of an `unless`.